Makefile

```diff
@@ -21,29 +21,26 @@ help: 		   										## Display this help text for Makefile
 .PHONY: upgrade
 upgrade:       										## Upgrade all dependencies to the latest stable versions
 	@echo "=> Updating all dependencies"
-	@if [ "$(USING_PDM)" ]; then $(PDM) update; fi
+	@if [ "$(USING_UV)" ]; then $(uv) lock --upgrade; fi
 	@echo "=> Dependencies Updated"
-	@$(PDM) run pre-commit autoupdate
+	@$(uv) run pre-commit autoupdate
 	@echo "=> Updated Pre-commit"
 
 # =============================================================================
 # Developer Utils
 # =============================================================================
-.PHONY: install-pdm
-install-pdm: 										## Install latest version of PDM
-	@curl -sSLO https://pdm.fming.dev/install-pdm.py && \
-	curl -sSL https://pdm.fming.dev/install-pdm.py.sha256 | shasum -a 256 -c - && \
-	python3 install-pdm.py && \
-	rm install-pdm.py
+.PHONY: install-uv
+install-uv: 										## Install latest version of uv
+	@curl -LsSf https://astral.sh/uv/install.sh | sh
 
 .PHONY: install
 install: clean										## Install the project, dependencies, and pre-commit for local development
-	@if ! $(PDM) --version > /dev/null; then echo '=> Installing PDM'; $(MAKE) install-pdm; fi
+	@if ! $(uv) --version > /dev/null; then echo '=> Installing uv'; $(MAKE) install-uv; fi
 	@if [ "$(VENV_EXISTS)" ]; then echo "=> Removing existing virtual environment"; fi
```

Comment:
i'm not sure what the rationale behind all the conditionality with venvs and tooling and whatever is, but i feel like it'd be overall easier if it...didn't exist? like to me the Makefile's only purpose is to define how the project-specific tooling ought to operate. If you're not using (uv in this case) the tooling, then you'd just not use the makefile?

The reason i bring it up here being there'd be no reason to have `detroy`/`clean` commands at all really.

Reply:
im not sure if this was from trying to support multiple things at once (pip, poetry) or not but it can die

Reply:
> The reason i bring it up here being there'd be no reason to have detroy/clean commands at all really.

What's the workflow you are thinking here?

Reply:
just that you **almost** dont even need a `make install`. by default `uv run pytest` will sync ahead of time. so i'm just not sure what cleaning out the venv would actually do.

Reply:
I'm mostly coming from my **very** simple makefiles from e.g. [cappa](https://github.com/DanCardin/cappa/blob/main/Makefile) which are **just** ways to ensure the command i run locally is exactly what's getting run in CI.

that, and it's really easy to see the total set of linters/formatters being applied to the project. by comparison, i've found the makefile setup (and pre-commit in particular) to be a confusing impediment